lms/djangoapps/completion/api/v1/views.py

```diff
@@ -0,0 +1,438 @@
+            specified, completion data is also returned for the requested block
+            types.  If any invalid fields are requested, a 400 error will be
+            returned.
+
+    **Returns**
+
+        * 200 on success with above fields
+        * 400 if an invalid value was sent for requested_fields.
+        * 403 if a user who does not have permission to masquerade as another
+          user specifies a username other than their own.
+        * 404 if the user is not enrolled in the requested course.
+
+        Example response:
+
+            {
+              "course_key": "course-v1:GeorgetownX+HUMX421-02x+1T2016",
+              "completion": {
+                "earned": 12.0,
+                "possible": 24.0,
+                "ratio": 0.5
+              },
+              "mean": 0.25,
+              "chapter": [
+                {
+                  "course_key": "course-v1:GeorgetownX+HUMX421-02x+1T2016",
+                  "block_key": "block-v1:GeorgetownX+HUMX421-02x+1T2016+type@chapter+block@Week-2-TheVitaNuova"
+                  "completion": {
+                    "earned: 12.0,
+                    "possible": 24.0,
+                    "ratio": 0.5
+                  }
+                }
+              ],
+              "sequential": [
+                {
+                  "course_key": "course-v1:GeorgetownX+HUMX421-02x+1T2016",
+                  "block_key": "block-v1:GeorgetownX+HUMX421-02x+1T2016+type@sequential+block@e0eb7cbc1a0c4000bec36b67e622c988",
+                  "completion": {
+                    "earned: 12.0,
+                    "possible": 12.0,
+                    "ratio": 1.0
+                  }
+                },
+                {
+                  "course_key": "course-v1:GeorgetownX+HUMX421-02x+1T2016",
+                  "block_key": "block-v1:GeorgetownX+HUMX421-02x+1T2016+type@sequential+block@f6e7ec3e965b48428197196acf3418e7",
+                  "completion": {
+                    "earned: 0.0,
+                    "possible": 12.0,
+                    "ratio": 0.0
+                  }
+                }
+              ]
+            }
+
+    This is a transitional implementation that uses the
+    edx-solutions/progress-edx-platform-extensions models as a backing store.
+    The replacement will have the same interface.
+    """
+    # pylint: enable=line-too-long
+
+    def get(self, request, course_id):
+        """
+        Handler for GET requests.
+        """
+        course_key = CourseKey.from_string(course_id)
+
+        # Return 404 if user does not have an active enrollment in the requested course
+        if not CourseEnrollment.is_enrolled(self.user, course_key):
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        try:
+            # Fetch the Aggregate comletions for the course
+            completion = self.get_queryset().filter(course_key=course_key)
+        except AggregateCompletion.DoesNotExist:
```

Comment:
Are you expecting DoesNotExist exceptions from the filter query? Filters usually return empty lists when they find nothing.